src/ConstructiveSolidGeometry/PointsAndVectors/Points.jl

```diff
@@ -24,6 +24,35 @@ struct CartesianPoint{T} <: AbstractCoordinatePoint{T, Cartesian}
     z::T
 end
 
+#Type promotion happens here
+function CartesianPoint(x::TX, y::TY, z::TZ) where {TX,TY,TZ}
+    eltypes = _csg_get_promoted_eltype.((TX,TY,TZ))
+    T = float(promote_type(eltypes...))
+    CartesianPoint{T}(T(x),T(y),T(z))
+end
+
+function CartesianPoint(x::TX, y::TY, z::TZ) where {TX<:Int,TY<:Int,TZ<:Int}
+    eltypes = _csg_get_promoted_eltype.((TX,TY,TZ))
+    T = float(promote_type(eltypes...))
+    CartesianPoint{T}(T(x),T(y),T(z))
+end
```

Comment:
I do not see the difference between these two...
The upper one should do exactly the same thing as the lower one, or am I missing something?

Reply:
If the second on is missing, CartesianPoint() goes directly to the struct, which therefor will be a CartesianPoint{Int64}

Reply:
Well, we introduced `CartesianPoint{Int}` to not be dominant in type promotion for other primitives.
With this method, `CartesianPoint{Int}` will always result in `CartesianPoint{Float64}` (because `float(Int64) = Float64`), which might cause volume primitives defined in `Float32` to become of type `Float64`.. I wonder why/if the tests fail.

Reply:
I think we only need the second method. Just use `<:Real` instead `<:Int`